application/templates/secretProviderClass.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if .Values.secretProviderClass.enabled }}
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: {{ .Values.secretProviderClass.name }}
+  namespace: {{ .Values.secretProviderClass.namespace }}
+spec:
+  provider: {{ .Values.secretProviderClass.provider }}
+  parameters:
+    {{- if .Values.secretProviderClass.vaultAddress }}
+    vaultAddress: {{ .Values.secretProviderClass.vaultAddress }}
+    {{- end }}
+    roleName: {{ .Values.secretProviderClass.roleName }}
+    {{- if .Values.secretProviderClass.objects }}
+    objects:
+    {{ toYaml .Values.secretProviderClass.objects | indent 4 }}
+    {{- end }}
+  {{- if .Values.secretProviderClass.secretObjects }}
+  secretObjects:
+  {{- range $value := .Values.secretProviderClass.secretObjects }}
+    - data: {{- toYaml $value.data | nindent 8 }}  
+      secretName: {{ $value.secretName }}
+      type: {{ $value.type }}
+    {{- end }}
+  {{- end }}
+  {{- end }}
```

Comment:
fix the indentation. remove extra spaces

Reply:
Isn't it supposed to be like this ?

Reply:
done